src/cpp/core/include/core/http/Response.hpp

```diff
@@ -273,8 +274,24 @@ class Response : public Message
       if (request.acceptsEncoding(kGzipEncoding))
          setContentEncoding(kGzipEncoding);
       
-      // set body from file
-      Error error = setBody(filePath, filter);
+      Error error;
+
+      // Apply padding to fix QT short response issues
+      if (filePath.mimeContentType() == "text/html")
+      {
+          std::string body = file_utils::readFile(filePath);
```

Comment:
For safety we should probably use exactly the same mechanism we currently use to read the file in `setBody` : https://github.com/rstudio/rstudio/blob/347477f0ab9ef6af430ea8a80b27a4a1d0b73ca0/src/cpp/core/include/core/http/Response.hpp#L224